File: backends/ts-hono/src/services/invitationService.ts
import { notFound } from "../lib/errors.js";
import { getCurrent, issueInvitation, findInvitationByToken } from "../repositories/invitationRepo.js";
import type { InvitationResult } from "../repositories/invitationRepo.js";
import type { Invitation } from "../db/schema.js";
import { config } from "../config.js";

export async function current(): Promise<InvitationResult> {
  const result = await getCurrent(config.app.frontendUrl);
  if (!result) throw notFound("invitation_not_found");
  return result;
}

export async function issue(): Promise<InvitationResult> {
  return issueInvitation(config.app.frontendUrl);
}

export async function findByToken(token: string): Promise<Invitation> {
  const inv = await findInvitationByToken(token);
  if (!inv) throw notFound("invitation_not_found");
  return inv;
}

File: backends/python-fastapi/app/services/invitation_service.py
from app.exceptions import not_found
from app.repositories.invitation_repo import InvitationRepository, InvitationResult


class InvitationService:
    def __init__(self, repo: InvitationRepository):
        self.repo = repo

    def current(self) -> InvitationResult:
        result = self.repo.get_current()
        if result is None:
            raise not_found("invitation_not_found")
        return result

    def issue(self) -> InvitationResult:
        return self.repo.issue()

    def find_by_token(self, token: str):
        inv = self.repo.find_by_token(token)
        if inv is None:
            raise not_found("invitation_not_found")
        return inv
